cli/cmd/test_helper.go

```diff
@@ -9,9 +12,29 @@ import (
 func diffCompare(t *testing.T, actual string, expected string) {
 	if actual != expected {
 		dmp := diffmatchpatch.New()
-		diffs := dmp.DiffMain(actual, expected, true)
+		diffs := dmp.DiffMain(expected, actual, true)
 		patches := dmp.PatchMake(expected, diffs)
 		patchText := dmp.PatchToText(patches)
 		t.Fatalf("Unexpected output:\n%+v", patchText)
 	}
 }
+
+func readGoldenTestFile(t *testing.T, testDirWithTrailingSlash string, goldenFileName string) string {
+	var fileData string
+
+	if goldenFileName != "" {
```

Comment:
i would fail the test if `goldenFileName` == ""

Reply:
There are some cases where a file name isn't provided by the test case in a table driven test and so we do not want to fail the test. For example:
```go
expectedStdOutResult := readGoldenTestFile(t, "testdata/", tc.stdOutGoldenFileName)
...
expectedStdErrResult := readGoldenTestFile(t, "testdata/", tc.stdErrGoldenFileName)
...
```
In one test case, we can have a test return data for `stdOut` but return nothing for `stdErr`. That test is considered valid because that is the intended functionality so we want to be able to load the golden file for `stdOut` and just return and an empty string for `stdErr` to be used to validate the test.

Reply:
> In one test case, we can have a test return data for stdOut but return nothing for stdErr. That test is considered valid because that is the intended functionality so we want to be able to load the golden file for stdOut and just return and an empty string for stdErr to be used to validate the test.

Probably the best way to address @siggy's feedback is to name the function "readOptionalTestFile" and document in a comment what happens when the filename is empty.